### ISSUE TITLE
Race condition in gocron when using irmago as dependency

### ISSUE THREAD

**Issue by woutslakhorst** [privacybydesign/irmago#268]
We use irmago as dependency and encounter a race condition when running our integration tests with `-race`

The detected race condition:
```
==================
WARNING: DATA RACE
Write at 0x00c0004a2bb8 by goroutine 64:
  github.com/privacybydesign/irmago.(*RevocationStorage).Load()
      /Users/wout.slakhorst/go/pkg/mod/github.com/privacybydesign/irmago@v0.11.0/revocation.go:779 +0x9bc
  github.com/privacybydesign/irmago.(*Configuration).ParseFolder()
      /Users/wout.slakhorst/go/pkg/mod/github.com/privacybydesign/irmago@v0.11.0/irmaconfig.go:191 +0xc3c
  github.com/nuts-foundation/nuts-node/auth/services/irma.GetIrmaConfig()
      /Users/wout.slakhorst/nuts/nuts-node/auth/services/irma/irmaconfig.go:51 +0x190
  github.com/nuts-foundation/nuts-node/auth/services/irma.defaultValidator()
      /Users/wout.slakhorst/nuts/nuts-node/auth/services/irma/validator_test.go:151 +0xf0
  github.com/nuts-foundation/nuts-node/auth/services/irma.TestService_VerifyVP.func1()
      /Users/wout.slakhorst/nuts/nuts-node/auth/services/irma/validator_test.go:59 +0x30
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.19.2/libexec/src/testing/testing.go:1446 +0x188
  testing.(*T).Run.func1()
      /opt/homebrew/Cellar/go/1.19.2/libexec/src/testing/testing.go:1493 +0x40

Previous read at 0x00c0004a2bb8 by goroutine 66:
  github.com/privacybydesign/irmago.(*RevocationStorage).updateAccumulatorTimes()
      /Users/wout.slakhorst/go/pkg/mod/github.com/privacybydesign/irmago@v0.11.0/revocation.go:546 +0x38
  github.com/privacybydesign/irmago.(*RevocationStorage).Load.func1()
      /Users/wout.slakhorst/go/pkg/mod/github.com/privacybydesign/irmago@v0.11.0/revocation.go:758 +0x2c
  runtime.call16()
      /opt/homebrew/Cellar/go/1.19.2/libexec/src/runtime/asm_arm64.s:478 +0x74
  reflect.Value.Call()
      /opt/homebrew/Cellar/go/1.19.2/libexec/src/reflect/value.go:368 +0x90
  github.com/go-co-op/gocron.callJobFuncWithParams()
      /Users/wout.slakhorst/go/pkg/mod/github.com/go-co-op/gocron@v1.14.0/gocron.go:106 +0x204
  github.com/go-co-op/gocron.(*executor).start.func1.2()
      /Users/wout.slakhorst/go/pkg/mod/github.com/go-co-op/gocron@v1.14.0/executor.go:92 +0xc0
  github.com/go-co-op/gocron.(*executor).start.func1()
      /Users/wout.slakhorst/go/pkg/mod/github.com/go-co-op/gocron@v1.14.0/executor.go:99 +0x2a0
```

**Comment by ivard** [comment]
I checked the other issues and I saw an older issue #57 that seems to indicate on something similar.

**Comment by stevenvegt** [comment]
Why cron? You can perhaps remove all this and just use a timer?

**Comment by ivard** [comment]
We don't use the library anymore that causes #57, so I closed that one and let's consider this as a new issue.

**Comment by ivard** [comment]
> Why cron? You can perhaps remove all this and just use a timer?

I think the issue is not related to `gocron`, but there is some kind of concurrency issue going on in the `RevocationStorage`. Using a timer won't resolve the actual issue, because there might always be concurrent calls to this storage.

**Comment by ivard** [comment]
If I understand it right you call `ParseFolder()` yourself in your unit tests? I think the issue is that you also spin up an IRMA server and that both invocations want to call the same global variables. That scenario wasn't considered to occur.

Can you check what happens if you cancel the `Scheduler` into the `Configuration` after you called `ParseFolder`? Then we know the source of the issue.

**Comment by woutslakhorst** [comment]
I think the problem lies in `revocation.go` 

On line 757-772 jobs are scheduled that access `rs.sqlMode` which is written to at 776-789. Since the scheduler is already started, concurrency happens.

**Comment by sietseringers** [comment]
Thanks @woutslakhorst, that was indeed the problem. Fix in #270.
